source/Outbox.Application/Extensions/DependencyInjection/OutboxExtensions.cs

```
@@ -0,0 +1,76 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using BuildingBlocks.Application.Extensions.DependencyInjection;
+using Energinet.DataHub.Core.App.Common.Extensions.DependencyInjection;
+using Energinet.DataHub.EDI.BuildingBlocks.Interfaces;
+using Energinet.DataHub.EDI.DataAccess.Extensions.DependencyInjection;
+using Energinet.DataHub.EDI.Outbox.Domain;
+using Energinet.DataHub.EDI.Outbox.Infrastructure;
+using Energinet.DataHub.EDI.Outbox.Interfaces;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Energinet.DataHub.EDI.Outbox.Application.Extensions.DependencyInjection;
+
+public static class OutboxExtensions
+{
+    /// <summary>
+    /// Add services required for creating and persisting outbox messages.
+    /// </summary>
+    public static IServiceCollection AddOutboxModule(
+        this IServiceCollection services,
+        IConfiguration configuration)
+    {
+        ArgumentNullException.ThrowIfNull(configuration);
+        services.AddBuildingBlocks(configuration)
+            .AddScopedSqlDbContext<OutboxContext>(configuration)
+            .AddScoped<BuildingBlocks.Domain.ExecutionContext>();
+
+        // DataRetentionConfiguration
+        services.AddTransient<IDataRetention, OutboxRetention>();
+
+        services
+            .AddNodaTimeForApplication();
+
+        services.AddTransient<IOutboxRepository, OutboxRepository>();
+        services.AddTransient<IOutboxClient, OutboxClient>();
+
+        return services;
+    }
+
+    /// <summary>
+    /// Add services required for processing and publishing outbox messages.
+    /// <remarks>Requires <see cref="AddOutboxModule"/> to be registered as well</remarks>
+    /// </summary>
+    public static IServiceCollection AddOutboxProcessor(
+        this IServiceCollection services)
+    {
```

Should we throw an exception if the dependencies from `AddOutboxModule` have not been registered?

---

☎️